test/NuGet.Core.Tests/NuGet.Protocol.Tests/Providers/PackageDetailsUriResourceV3ProviderTests.cs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using Newtonsoft.Json.Linq;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.Protocol.Tests.Providers
+{
+    public class PackageDetailsUriResourceV3ProviderTests
+    {
+        private const string _resourceType = "PackageDetailsUriTemplate/5.0.0";
+
+        private static readonly SemanticVersion _defaultVersion = new SemanticVersion(0, 0, 0);
+
+        private readonly PackageSource _packageSource;
+        private readonly PackageDetailsUriResourceV3Provider _target;
+
+        public PackageDetailsUriResourceV3ProviderTests()
+        {
+            _packageSource = new PackageSource("https://unit.test");
+            _target = new PackageDetailsUriResourceV3Provider();
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenResourceDoesNotExist_ReturnsNoResource()
+        {
+            var resourceProviders = new ResourceProvider[]
+            {
+                CreateServiceIndexResourceV3Provider(),
+                _target
+            };
+            var sourceRepository = new SourceRepository(_packageSource, resourceProviders);
+
+            var result = await _target.TryCreate(sourceRepository, CancellationToken.None);
+
+            Assert.False(result.Item1);
+            Assert.Null(result.Item2);
+        }
+
+        [Theory]
+        [InlineData("foo")]
+        [InlineData("../somepath")]
+        [InlineData(@"C:\packages\{id}\{version}\index.html")]
+        [InlineData("http://unit.test/packages/{id}/{version}")]
+        [InlineData("file:///my/path")]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData("  \t\n")]
+        public async Task TryCreate_WhenResourceHasInvalidAbsoluteUri_ReturnsNoResource(string uri)
+        {
+            var serviceEntry = new RawServiceIndexEntry(uri, _resourceType);
+            var resourceProviders = new ResourceProvider[]
+            {
+                CreateServiceIndexResourceV3Provider(serviceEntry),
+                _target
+            };
+            var sourceRepository = new SourceRepository(_packageSource, resourceProviders);
```

Comment:
I was expecting you would get the resource through the source repository.

Reply:
I am, on the next line. `TryCreate`. This is mimicking `RepositorySignatureResourceProviderTests`.